web/source/kmwbase.ts

```diff
@@ -181,15 +181,26 @@ namespace com.keyman {
       this.touchAliasing = this.util.device.touchable ? this.domManager.touchHandlers : this.domManager.nonTouchHandlers;
     }
 
+    /**
+     * Reset context when entering or exiting the active element.
+     * Will also trigger OSK shift state / layer reset.
+     **/
+    pageFocusHandler = () => {
+      if(!this.uiManager.isActivating && this.osk?.vkbd && this.util.device.touchable) {
+        this.core.keyboardProcessor._UpdateVKShift(null);
+      }
+      return false;
+    }
```

Comment:
Note lines 189 & 190 - they differ slightly from the 16.0 version in order to keep this change a little closer to the original stable-15.0 code.

Reference https://github.com/keymanapp/keyman/pull/6902#discussion_r916426897 and its chain for the reasoning.